<head>
  <meta charset='utf-8' />
  <link href='../normalize.css' rel='stylesheet' />
  <link href='../styles.css' rel='stylesheet' />
  <title>Notes from Systems Biology Winter School | Shorts by Vivek Rai</title>
</head>
<body>
  <div class='container'>
    <div class='banner no-print'>
      <div class='banner-title'>
        <a href='https://vivekiitkgp.github.io/shorts'>Shorts</a>
      </div>
      <div class='banner-subclause'>
        Musings of a free mind, treasured recollections <br> &amp; <br> visions of a distant future
      </div>
      <div class='archive'>
        <a href='../places.html'>places,</a>
        <a href='../books.html'>books,</a>
        <a href='../science.html'>science,</a>
        <a href='../programming.html'>programming,</a>
        <br>
        <a href='../notes.html'>notes,</a>
        <a href='../unfinished.html'>unfinished,</a>
        <a href='../about.html'>about me</a>
      </div>
    </div>
    <div class='content'>
      <article>
        <div class='metadata'>
          <div class='category'>
            <a href='../science.html'>
              science
            </a>
          </div>
          <div class='timestamp'>
            06 Jan 2016
          </div>
        </div>
        <div class='title'>
          <a href='../science/winter-school.html#main'>
            Notes from Systems Biology Winter School
          </a>
        </div>
        <div class='post'>
          <p>This winter I attended the fourth session of <a href="https://www.icts.res.in/program/current/details/361/">Winter School on Quantitative
Systems Biology</a> at <a href="https://icts.res.in">International Centre for Theoretical Sciences (ICTS)</a>,
Bangalore. The school is organized by the ICTS, Bangalore and <a href="https://www.ictp.it">ICTP,
Italy</a> in
succession as a part of their ICTP-ICTS Programme in Biology since 2012.</p>
<p>It was an extraordinarily fun-filled learning experience. Here is a brief
summary of my impressions of the school.
<br>
<br>
<small><em>Caution: Not so short. Also, the post may be frequently updated.</em></small> </p>
<hr />
<h3>TL;DR</h3>
<p>The winter school provides great learning experience and an excellent platform
to make connections and get to know people. It has a strong theme yet remains
broad enough to be of interest to at an interdisciplinary audience. More
importantly, you'd enjoy working in a mix company of people from different
backgrounds brainstorming and working together towards solving a similar problem
of interest. And not to mention, it is a humbling experience to witness the
diversity and scale of research being carried out by people around the world.</p>
<p>I heartily recommend Winter School on Quantitative Systems Biology to anyone who
feels moderately excited with these prospects.</p>
<p>A comprehensive log of daily lectures, resources and notes are
<a href="https://github.com/vivekiitkgp/qsb-school-diary.git">online</a>. Also, check out
the YouTube <a href="">channel</a> of ICTS for video recordings.</p>
<hr />
<h3>Theme</h3>
<p>Typically, each school has a central theme and accordingly distinguished
speakers from around the world in the concerned fields are invited.
Additionally, before each session consists of an extra pre-school component
where instructors introduce everyone to the relevant pre-requisite concepts for
the main school. This was an effective strategy because the main school turns
out to be quite intensive with series of lectures, research talks, poster
presentation, and discussion session lined up in succession. We even had a small
project component towards the end.</p>
<p>The description of QSB2015 reads:</p>
<blockquote>
<p>QSB2015 is centered around <strong>bacteria</strong>, the simplest known forms of life. The
focus is on the basics of cellular life and the principles thereof. The
emphasis is on the structures and processes that allow bacteria to survive,
reproduce, evolve and form communities in their environment. This is a setting
that is fertile for quantitative work, both theoretical and experimental, and
for grasping some universals in biology.</p>
</blockquote>
<h3>Participants</h3>
<p>To my surprise, most (~80%) of the <a href="">school participants</a> were either physicists or
mathematicians. Being in a crowd of this composition, as a biologist, provides
you an unique insight of how others perceive and understand the key concepts
that we often take for granted. How to reduce the system, formulate the problem,
model the system etc., are a few key questions that are well understood in
company of people who, generally, are better at it.</p>
<h3>Subjects - Overview</h3>
<p>A gross categorization:</p>
<h4>Pre-school</h4>
<ul>
<li>Numbers in Biology                (<a href="">Mukund Thattai</a>)</li>
<li>Mathematical modeling of processes (<a href="">Sandeep Krishna</a>)</li>
<li>Ecology and Evolution             (<a href="">Deepa Agashe</a>)</li>
<li>Primer on Biology                 (<a href="">Supreet Saini</a>)</li>
<li>Metagenomics                      (<a href="">Nagasuma Chandra</a>)</li>
</ul>
<h4>Main school</h4>
<ul>
<li>Bacterial cell physiology and growth <br> (<a href="">Matthew Scott</a>, <a href="">Suckjoon Jun</a>)</li>
<li>Metabolism, regulation and response to the environment <br> (<a href="">Tom Khulmann</a>)</li>
<li>Structure and dynamics of biochemical networks <br>
(<a href="https://en.wikipedia.org/wiki/Gary_Stormo">Gary Stormo</a>, <a href="">Nathan Price</a>, <a href="">Nagasuma Chandra</a>, <a href="http://chaos.c.u-tokyo.ac.jp/">Kunihiko Kaneko</a>)</li>
<li>Bacterial ecology <br> (<a href="">Tzachi Pilpel</a>)</li>
<li>Evolution <br> (<a href="http://www.ieu.uzh.ch/wagner/">Andreas Wagner</a>, <a href="">Matthias
Heinemann</a>, <a href="">Tzachi Pilpel</a>)</li>
<li>Energy, information and computation in cells <br> (<a href="">Pankaj Mehta</a>)</li>
</ul>
<h3>Topics</h3>
<h4>Bacterial physiology</h4>
<p>I had a notion that a lot of work* has been done in mainstream microbiology
causing people to shift away from it to modern buzzwords in biology. However, in
this school, I realized that it <em>still</em> is the bread and butter for many
scientists and people are constantly pushing the boundaries by coming up with
interesting work.</p>
<p><strong>Matthew Scott</strong> walked us through through an entire theory of quantitative
bacterial physiology using the classic case studies from 60s. The key question
being - How do the growth, size, cell cycle, macromolecular composition and
morphology of a cell coordinate? Can we predict anything? In an answer to these
questions, concepts of growth law, proteome partitioning, generation time versus
division time puzzle, along with comprehensive quantitative models of growth
were discussed.</p>
<p><strong>Suckjoon Jun</strong> built upon the content discussed by Matt Scott and described his
work on <a href="http://jun.ucsd.edu/mother_machine.php"><em>mother machine</em></a> and how they have use it to conduct experiments. The
key idea of his lectures were to infer the physiological and metabolic control
parameters of a bacterial cell that can explain the phenomenon of size and
growth homeostasis. They have proposed <strong>adder theory</strong> as an explanation to
cell size homeostasis and hypothesize growth \( (\lambda) \) and size \((\Delta)\) as two independent
and sufficient control parameters for bacterial physiological regulation.</p>
<p>* <small> Thanks to Cooper, Monod, Helmstetter, Schaechter et al. in 50s. </small></p>
<h4>Metabolism and biochemical networks</h4>
<p><strong>Nathan Price</strong> discussed some of standard techniques and algorithms used in
metabolic network model construction and analysis. Apparently, creating a full
genome scale metabolic network is significantly challenging and humongous task
typically involving data collection, literature mining, processing, analysis,
validation, and curation as key steps. At each step, there are tricky of dealing
with missing values, right scoring function, and database to use. Once you have
reconstructed a network, next step is to find out its valid states. A popular
technique used for network reconstruction and analysis is COnstraint-Based
Reconstruction and Analysis (COBRA) that employs physicochemical and biological
constraints to limit the space of phenotypic states of a network in a given
condition. Naturally, here <em>more data means better model</em>.</p>
<p>&ensp;&ensp;Once we have constructed a valid metabolic network, the
next step could be to integrate it with the regulatory network. The enhanced
network can now be used in retrospect to improve gene annotation, identify new
metabolites and thereby improve itself. It also incorporates an amazing
predictive power that may guide new biological hypothesis development. Infact,
anything that you can do with a normal network can be done with this network,
added with a biological significance.</p>
<p>On the other hand, <strong>Mathhias Heinemann</strong> took us to metabolism on a single cell
level and discussed how cells regulate their metabolism. He proposed an elegant
<em>flux-sensing mechanism</em> used by the cells to continuously measure their outside
environment and carry out internal changes. A fantastic study by his group
demonstrated the phenotypic bistability of <em>E.coli</em>'s central carbon metabolism
and how it can be achieved by a simple use of few <em>flux-sensors</em> along the
network.</p>
<p>Incidentally, the project component I was involved in was also based on
metabolic network analysis. Perhaps, that should go in my next post.</p>
<h4>Evolution</h4>
<p>Since the seminal work of Darwin and Wallace almost two centuries ago, evolution
is still one of the hottest studied areas. The school witnessed several lectures
and research talks on topics like evolutionary dynamics, molecular evolution
with interesting theories put forward frequently.</p>
<p>At ETH Zurich, Andreas Wagner is trying to understand the origin of innovations,
complexity and diversity of life using <em>systems approach</em>. The primary question
being, "<em>Are there any principles of innovability in nature that govern the
arrival of fittest? Or is it just random mutation and selection over
evolutionary period?</em>".</p>
<blockquote>
<p>Innovation is the origin of a major qualitative feature that cannot a
It can occur in following forms: <br>
1. Metabolic innovation <br>
2. Regulatory innovation <br>
3. New macromolecule</p>
</blockquote>
<p>However, one must carefully distinguish <em>innovation</em> with <em>adaptation</em> to
understand the question correctly.</p>
<p>&ensp;&ensp;His group is looking for answers by probing the metabolic phenotype and
genotype space and their relation in a network. The analysis yield few
curious results like - </p>
<ol>
<li>Two genotypes can differ (determined by a distance function D) strikingly
despite having same phenotype.</li>
<li>A genotype can have quite diverse neighborhood. </li>
</ol>
<p>A consequence of these two properties is that it is really <em>not that difficult</em>
for a genotype to acquire a new, unique phenotype. Combine with regulatory
phenotype space, protein sequence space \((20^n)\), RNA sequence space
\((4^n)\) etc., you have a multi-dimensional space to possibly generate any
combination with very few step movements across the hyperspace. More info in his
papers.</p>
<p>If this seems too computational to your taste, <a href="http://longitude.weizmann.ac.il/"><strong>Tzachi Pilpel</strong></a> discussed how
evolution occurs at the level of gene expression and related processes with
great experimental effort. A series of interesting papers on how transcription
and translation networks may have evolved. The fact that even <em>tRNA abundance</em>
could be optimized for, and can be used as an active means of control is
beautiful.</p>
<p>Another exciting set of experiments were done by <strong>Tom Kuhlmann</strong> whose group
wanted to understand the role of transposons in evolution. The experiments
beautifully showed transposons in action, in <em>real-time</em> using a rather simple
plasmid construct. Quite a few conclusions follow from these experiments that
I encourage anyone to look up in his papers.</p>
<p>Finally, the research talk by <strong>Kunihiko Kaneko</strong>, professor at Un</p>
<h4>Energy, information and computation in cells</h4>
<p>How do cells perform computation?(*) And what is it that requires energy? Is
everything inside a cell simply driven by thermodynamics and a careful design of
free energy cascade? Do these processes have any fundamental limitations that
biological systems have to obey?&ensp;&ensp;&ensp;&ensp;</p>
<p>As it turns out, <em>reality</em> is not so simple.</p>
<p>*<small>Sensing the chemical gradient, Random walks, <a href="http://www.nature.com/news/how-brainless-slime-molds-redefine-intelligence-1.11811">Slime molds</a> etc.</small> </p>
<p><strong>Pankaj Mehta</strong>, a physicist from Boston University works on elegant theories
to answer these questions. They key motivation of his talks was to investigate
<em>no-go</em> theories in biology, and their possible use in design of synthetic
cellular modules and circuits. He discussed two classical papers, <em>Kinetic
Proofreading</em> by Hopfield et al. (1994), and <em>Physics of Chemoreception</em> by Burg and
Purcell (1975). The two papers provide an excellent discussion of how cells employ
active processes to increase their specificity. Turns out, that's where you
break the <em>time-reversal symmetry</em> and hence <em>have to</em> consume energy.</p>
        </div>
        <div class='article-end'>
          
            Published by Vivek Rai
          
        </div>
        <div class='share'>
          <a class='twitter-share-button' data-dnt='true' data-via='vivek_ziel' data-text='Notes From Systems Biology Winter School: Shorts by Vivek Rai' href='https://twitter.com/share' data-hashtags='science'>tweet |&nbsp;</a>
          <a class='twitter-follow-button' data-show-screen-name='false' data-dnt='true' data-show-count='false' href='https://twitter.com/vivek_ziel'>follow @vivek_ziel</a>
        </div>
      </article>
    </div>
  </div>
  <script src='//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-MML-AM_CHTML'></script>
  <script>
    !function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');

  </script>
</body>