<!doctype html><html lang=en>
<head>
<meta charset=utf-8>
<meta name=author content>
<meta name=robots content="index,follow">
<meta name=referrer content="no-referrer">
<meta name=viewport content="width=device-width,initial-scale=1">
<link rel="shortcut icon" href=https://raivivek.in/img/favicon.ico type=image/x-icon>
<link rel=canonical href=https://raivivek.in/2016/08/open-data-science-source-opensnp.org/>
<title>Open {data, science, source} - OpenSNP.org | Shorts</title>
<link rel=stylesheet href=https://raivivek.in/style.min.css>
<script src=//instant.page/3.0.0 type=module defer integrity=sha384-OeDn4XE77tdHo8pGtE1apMPmAipjoxUQ++eeJa6EtJCfHlvijigWiJpD7VDPWXV1></script>
</head>
<body>
<div class=c-banner>
<div class=c-banner-title>
SHORTS
</div>
<div class=c-banner-subclause>
<i>☘☘☘ by Vivek </i>
</div>
<div class=c-nav-bar>
<a href=https://raivivek.in/>home</a>
<a href=https://raivivek.in/science>science</a>
<a href=https://raivivek.in/code>code</a>
<a href=https://raivivek.in/scribbles>scribbles</a>
<a href=https://raivivek.in/books>books</a>
<a href=https://raivivek.in/about>about</a>
</div>
</div>
<article>
<header>
<h2>Open {Data, Science, Source} - OpenSNP.org</h2>
<p class=subtitle> First Chapter, Best Summer Ever! </p>
</header>
<h2 id=april-2016>April, 2016&nbsp;<a class=anchor href=#april-2016></a> </h2>
<p>“What’s the worst that can happen? Not like it will be different this time.
some sleep”, I told myself as I prepared to brace the bad news. The clock moves
closer to midnight with every passing moment, but I am too tired to think
about it anymore.</p>
<p>About half an hour into my sleep, I am startled by a loud phone call. I check
time. It&rsquo;s 12.30 am. The results must have been announced already at the
midnight. Anxious to get it over with, I recall my own words from earlier and
prepare to face the worst. This year was my third attempt at <a href=https://summerofcodewith.google.com>Google Summer of
Code</a>.</p>
<p>But, this year, I was selected.</p>
<h2 id=august-2016>August, 2016&nbsp;<a class=anchor href=#august-2016></a> </h2>
<figure>
<label for=marginnote-a161a459ccd5a1e64d110c06ad23b419-0 class=margin-toggle>⊕</label>
<input type=checkbox id=marginnote-a161a459ccd5a1e64d110c06ad23b419-0 class=margin-toggle>
<span class=marginnote>
<a href=http://creativecommons.org/licenses/by-nc-nd/3.0/>
Google
</a>
</span>
<img src=/images/GSoC2016Logo.png>
</figure>
<p>It is the <a href=https://developers.google.com/open-source/gsoc/timeline>final week</a> of Google Summer of Code. Participating students are
required to present their work and submit a final wrap up post about their
project as the last task before final evaluation.</p>
<p>This blog post is my submission.</p>
<p>Through the summers, I have
been working on my proposed project &lsquo;<a href=https://summerofcode.withgoogle.com/organizations/5693436329984000/#4637393399316480>Linking Phenotypes to
Genotypes</a>&rsquo;
for <a href=https://opensnp.org>OpenSNP</a> under the umbrella of <a href=https://en.wikipedia.org/wiki/Open_Bioinformatics_Foundation>Open Bioinformatics
Foundation
(OBF)</a>. This
post is also a personal account of my experiences besides being a suitably
technical introduction to my project.</p>
<p>So, by the end of this post, I hope to provide everyone with a clear idea of what
I have been working on and why it matters, describe my <a href=https://trello.com/b/ubBNKnmT/opensnp>progress</a> on the project,
and share the marvelous experience of working with the OpenSNP team.</p>
<h2 id=project>Project&nbsp;<a class=anchor href=#project></a> </h2>
<blockquote cite>
<p> <p>Currently, OpenSNP portal has a decent database collection of <a href=https://en.wikipedia.org/wiki/Genotype>genotypes</a> (3000+ users), <a href=https://en.wikipedia.org/wiki/Phenotype>phenotypes</a>, <a href=https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism>SNPs</a> and
associated literature information all available in an easily searchable way.
There is, however, a lot of scope in improving the impact and reach of the
project by modernizing the interface, studying the aggregated data, and
providing intuitive analysis to the users to allow for quicker yet improved and
comprehensive understanding. For example, OpenSNP portal currently displays the
<a href=https://en.wikipedia.org/wiki/Allele>allele</a> frequency of
a genotype - a valuable insight into how rare your mutation is.</p>
<p>This proposal seeks to work on the second problem of bridging different
components to allow for better interactivity and information conveyance.</p>
</p>
<footer></footer>
</blockquote>
<p>In line with the proposition set up in the introduction, my project aims to
<a href=https://github.com/openSNP/snpr/blob/master/ROADMAP.md#long-term-6-months>Link the SNPs with
Phenotypes in the
portal</a>.</p>
<p><em>What do I mean by linking?</em></p>
<p><a href=https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism>Single Nucleotide Polymorphisms (SNPs)</a> (pronounced
<em>snips</em>), simply, are nucleotide changes in DNA that occur with
a certain percentage of a population.
<span>
<label for=sidenote-a161a459ccd5a1e64d110c06ad23b419-2 class="margin-toggle sidenote-number"></label>
<input type=checkbox id=sidenote-a161a459ccd5a1e64d110c06ad23b419-2 class=margin-toggle>
<span class=sidenote><a href=https://en.wikipedia.org/wiki/Mutation>Mutations</a> in DNA arise
all the time due to intrinsic stochasticity of biological processes. However, most of
these changes are
<a href=https://en.wikipedia.org/wiki/Neutral_mutation><em>neutral</em></a> and do not
benefit or harm the organism in any way, while others are self-corrected by
the cell. Rarely, a few variations occur in critical regions of the genome
(a gene that codes an essential protein, for example) and manage to escape
correction. This variation (including the phenotype or disease it may be
responsible for!) could then propagate through the population via sexual
recombination.</span>
</span></p>
<p>They can affect how humans develop diseases and respond to pathogens, chemicals, drugs,
vaccines, and other agents. A set of SNPs for a single individual can also act
as a unique signature useful in identification and forensics. Naturally,
figuring out the possible downstream effects of an SNP is indeed useful. And
that’s exactly what we mean by linking SNPs to phenotypes.
<span>
<label for=sidenote-a161a459ccd5a1e64d110c06ad23b419-3 class="margin-toggle sidenote-number"></label>
<input type=checkbox id=sidenote-a161a459ccd5a1e64d110c06ad23b419-3 class=margin-toggle>
<span class=sidenote>See <a href=https://en.wikipedia.org/wiki/SNP_genotyping>SNP Genotyping</a> for methods to detect SNPs.</span>
</span></p>
<p>However, a word of caution here. Figuring out the <em>causal</em> effects of a
particular SNP in reality a significantly difficult problem and a major topic
of study in the field of computational biology and genetics.
<span>
<label for=marginnote-a161a459ccd5a1e64d110c06ad23b419-4 class=margin-toggle>&#8853;</label>
<input type=checkbox id=marginnote-a161a459ccd5a1e64d110c06ad23b419-4 class=margin-toggle>
<span class=marginnote>
<a href=https://en.wikipedia.org/wiki/Genome-wide_association_study>Genome-wide association studies (GWAS)</a>
</span>
</span></p>
<p>Our current work instead builds previously published findings available to us
as references and <a href=https://en.wikipedia.org/wiki/Text_mining>mines</a> SNP to
phenotype associations reported in the results to perform a <a href=https://en.wikipedia.org/wiki/Meta-analysis>meta-anlaysis</a>.</p>
<h2 id=motivation>Motivation&nbsp;<a class=anchor href=#motivation></a> </h2>
<figure>
<label for=marginnote-a161a459ccd5a1e64d110c06ad23b419-5 class=margin-toggle>⊕</label>
<input type=checkbox id=marginnote-a161a459ccd5a1e64d110c06ad23b419-5 class=margin-toggle>
<span class=marginnote>
<a href=https://github.com/openSNP/Community-Design/tree/master/openSNP/RGB>
OpenSNP
</a>
</span>
<img src=/images/opensnp.png>
</figure>
<p><a href=https://opensnp.org>OpenSNP.org</a> has a single purpose — to act as a free,
open-data repository by collecting personal genomics data into the public
domain. The web-portal, thus, is the critical component of the infrastructure.
From my understanding, I can imagine two immediate benefits of this project.</p>
<ul>
<li>Improve the overall user-experience by making to easier to understand possible
associations between an SNP to a phenotype and vice-versa.</li>
<li>Build a semantic data structure (or network) of SNPs and Phenotypes linked
based on literature information.</li>
</ul>
<p>Although the second point is not the obvious goal of this project, I believe it
could potentially serve as a useful data source to openSNP users in future.</p>
<h2 id=progress>Progress&nbsp;<a class=anchor href=#progress></a> </h2>
<p>The main components of the project were: (<a href=https://github.com/vivekiitkgp/snpr/commits/master>commits on GitHub</a>)</p>
<ul>
<li>Schema updates, database model, building association table - ✓</li>
<li>Updating the controllers and views - ✓</li>
<li>Algorithm to search and score literature for phenotypes - ✓</li>
<li>Writing the worker script - ✓</li>
<li>Writing tests for everything above - <em>in progress</em></li>
<li>Executing Rake tasks - <em>later</em></li>
</ul>
<p>In the first step, I familiarized myself with Rails and ActiveRecord pattern before
moving to database associations. A useful strategy to learn fast was to set up
dummy examples. For example, using <code>rails console</code>, one could examine arbitrary
objects, query database, or modify objects. It became super easy to understand
concepts like foreign key, indexing, migration and debug errors. Through this,
I could learn how pieces of code come together to create a coherent
architecture, like the MVC pattern. — exciting start!</p>
<p>Next step was to write code to score phenotype matches across the references
corresponding to each SNP and rank the possible matches. The first strategy was
to try a simple frequency lookup. While this sounds easy and attractive
option, it is a <em>naive</em> solution, and there are some serious limitations
to it. For instance:</p>
<ul>
<li>OpenSNP’s <a href=https://opensnp.org/phenotypes>phenotypes table</a> is populated manually by users. The added phenotypes
often do no correspond to the phenotypes as reported in references. Hence,
there will be no match.</li>
<li>Since we are not searching through the full text of the references but only through
title and summary (for some papers, if available), it creates an issue
since many references do not name a phenotype in the title to avoid coming as too
strong when studies are only suggestive.</li>
</ul>
<p>All in all, you end up with too few phenotypes detected for a single SNP even
though it has tons of references. We discussed this and thought of several
ideas - use additional data aggregation tools (for example,
<a href=http://myvariant.info>myvariant.info</a>), use a standard list of phenotypes, or
even <em>better</em> create a phenotype network and leverage phenotype associations.</p>
<figure>
<label for=marginnote-a161a459ccd5a1e64d110c06ad23b419-6 class=margin-toggle>⊕</label>
<input type=checkbox id=marginnote-a161a459ccd5a1e64d110c06ad23b419-6 class=margin-toggle>
<span class=marginnote>
<strong>OpenSNP.org dashboard with SNP to Phenotype linkage prototype. </strong>
The SNP info dashboard showing current columns of genotype and allele frequency along with newly implemented (for demo purposes!) list of suggestive phenotypes.
</span>
<img src=/images/pheno_list.png>
</figure>
<p>However, for the sake of simplicity and as a first milestone, we proceeded with the
initial naive approach hoping to improve it in future iterations. The final
logic was packed into a <code>Sidekiq</code> <a href=https://github.com/vivekiitkgp/snpr/blob/master/app/workers/link_snp_phenotype.rb>worker
script</a>
— so far so good.</p>
<p>The next critical task was to write extensive unit tests and document all of the
implemented changes. This is what I have been mainly doing in the last
few weeks of the project. In the remaining time, I hope to continue testing and
improving on the worker script. I obtained an actual data dump from the
production database that I can use actually to check and see the performance.
Especially because it’s a per SNP operation and there are thousands of it. So, it
would be crucial to see how much additional overhead it adds to the production
server.</p>
<p>Besides, during this process, I came across several aspects of the code that could be
improved. For example, Phenotype table did not have a <a href=https://github.com/openSNP/snpr/issues/304>unique key
constraint</a> on the <code>characteristic</code>
column. So, you could have two (or more!) rows describing <em>hair color</em> or
<em>height</em>.</p>
<p>Overall, I would not say that I delivered entirely on each deliverable, but we
definitely moved in a forward direction. Several new challenges came up, but
we hope to continue working on their resolution - after and beyond the scope of GSoC.</p>
<h2 id=journey>Journey&nbsp;<a class=anchor href=#journey></a> </h2>
<p>I cannot thank my mentors enough for their continuous effort throughout the
project. <a href=https://twitter.com/PhilippBayer>Philipp</a> and
<a href=https://twitter.com/gedankenstuecke>Bastian</a> are one of the most enthusiastic
and voracious readers I have met so far. Thanks to an excellent stroke of luck,
I got to meet Bastian at BOSC 2016 in Orlando, Florida. More about it soon!</p>
<figure>
<label for=marginnote-a161a459ccd5a1e64d110c06ad23b419-7 class=margin-toggle>⊕</label>
<input type=checkbox id=marginnote-a161a459ccd5a1e64d110c06ad23b419-7 class=margin-toggle>
<span class=marginnote>
<strong>A lovely card along with cute OpenSNP stickers! </strong>
A lovely card with a personal message and free OpenSNP stickers as surprise from Bastian.
</span>
<img src=/images/bastian-note.jpg>
</figure>
<p>Last but not the least, a big thanks to Google for the amazing program! Also,
thanks to <a href=https://mateusjabour.github.io/>Mateus</a> and
<a href=https://sxibolet.2pitau.org>Graham</a>, the other summer of code students for
OpenSNP, for providing an excellent company throughout.</p>
</article>
<div class=footer>
<div class=footer-content>
<div>
<ul>
</ul>
<a href=# class="pure-menu-link pull-right" id=gototop-btn>↑↑</a>
</div>
</div>
</div>
</body>
</html>