<!doctype html><html lang=en><head><meta charset=utf-8><meta name=author content><meta name=robots content="index,follow"><meta name=referrer content="no-referrer"><meta name=viewport content="width=device-width,initial-scale=1"><link rel="shortcut icon" href=https://raivivek.in//favicon.ico type=image/x-icon><link rel=canonical href=https://raivivek.in/2014/05/introduction-to-sequenceserver/><title>Introduction to Sequenceserver | Shorts</title>
<link rel=stylesheet href=https://raivivek.in/style.min.css><script src=//instant.page/3.0.0 type=module defer integrity=sha384-OeDn4XE77tdHo8pGtE1apMPmAipjoxUQ++eeJa6EtJCfHlvijigWiJpD7VDPWXV1></script></head><body><div class=c-banner><div class=c-banner-title>SHORTS</div><div class=c-banner-subclause><i>by Vivek</i></div><div class=c-nav-bar><a href=https://raivivek.in/>home</a>
<a href=https://raivivek.in/science>science</a>
<a href=https://raivivek.in/code>code</a>
<a href=https://raivivek.in/scribbles>scribbles</a>
<a href=https://raivivek.in/books>books</a>
<a href=https://raivivek.in/about>about</a></div></div><article><header><h2>Introduction to Sequenceserver</h2><p class=subtitle>Easy BLASTing with bespoke interface</p></header><p>Contributing to open source is quite an exciting journey. I embarked
upon this path recently with few contributions to <a href=http://www.sequenceserver.com/>SequenceServer - A
BLAST tool</a>, a bioinformatics software
built around BLAST+ that lets you setup a custom BLAST+ server to perform BLAST queries
with your own database locally through a web interface.</p><p>For those who are not familiar with <strong>BLAST</strong> or Basic Local Search Alignment Tool
<span><label for=sidenote-0 class="margin-toggle sidenote-number"></label>
<input type=checkbox id=sidenote-0 class=margin-toggle>
<span class=sidenote><a href=https://blastalgorithm.com>Altschul et al. (1990)</a></span>
</span>, it is an algorithm which allows
quick comparison of biological sequence information which are long strings
consisting of amino acids (in case of proteins) or nucleotides (in case of
nucleic acids). It is routinely used by biologists all over the world to
study new organisms, discover genes, and find similarity between
proteins among many applications. One can see a working instance of the tool
at <a href=http://antgenomes.org/blast>Ant Genomes</a>, maintained by <a href=http://www.sbcs.qmul.ac.uk/staff/yannickwurm.html>Yannick
Wurm</a>. Dr. Wurm is a
professor at the Queen Mary University of London and spearheads the project.</p><p>The program has already completed its major part of development and a
stable version is already available for use. However, there are many pending
<a href=https://github.com/yannickwurm/sequenceserver/issues>issues</a> and
feature requests that could vastly improve the user experience and overall functionality of the tool.
Seeing this, I got in touch with the maintainers on GitHub and submitted
three pull requests fixing a few bugs and implementing a new feature.</p><p>While it was not a difficult task, I felt quite motivated with this small experience
of working with the community, engaging with the developers, and obviously
working with Git. Also, because SequenceServer is implemented in Ruby, this
little exercise gave me an interesting opportunity to brush up my knowledge
of Ruby and understanding the code. Ruby is a concise and appealing language
which is not very tough to understand once you are in with the flow.</p><p>Anyway, I plan to continue my contribution to SequenceServer for quite some
time and also encourage you to take a look if you are even just a little bit
of biology enthusiast. I will talk more about BLAST algorithm in my next
post.</p></article><div class=footer><div class=footer-content><div><ul></ul><a href=# class="pure-menu-link pull-right" id=gototop-btn>↑↑</a></div></div></div></body></html>